hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/block/DeletedBlockLogImpl.java

```diff
@@ -121,6 +120,33 @@ public DeletedBlockLogImpl(ConfigurationSource conf,
             containerManager, this.scmContext, metrics, scmCommandTimeoutMs);
   }
 
+  @Override
+  public List<DeletedBlocksTransaction> getFailedTransactionsBatch(
+      int batchSize, long startTxId) throws IOException {
+    List<DeletedBlocksTransaction> failedTXs = new ArrayList<>();
+
+    lock.lock();
+    try {
+      try (
+          TableIterator<Long, ? extends Table.KeyValue<Long, DeletedBlocksTransaction>> iter =
+              deletedBlockLogStateManager.getReadOnlyIterator()) {
+
+        iter.seek(startTxId);
+
+        while (iter.hasNext() && failedTXs.size() < batchSize) {
+          DeletedBlocksTransaction delTX = iter.next().getValue();
+          if (delTX.getCount() == -1) {
+            failedTXs.add(delTX);
+          }
+        }
+      }
+    } finally {
+      lock.unlock();
+    }
+
+    return failedTXs;
+  }
+
```

Comment:
Why do we need this additional method? The same thing can be achieved with the existing `getFailedTransactions` method.

Reply:
Done